### PR TITLE
fix compilation warnings for gcc unit tests

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: check.py [hunk 1]
```diff
@@ -306,7 +306,7 @@ def run_gcc_tests():
             # build the C file separately
             libpath = os.path.join(os.path.dirname(shared.options.binaryen_bin),  'lib')
             extra = [shared.NATIVECC, src, '-c', '-o', 'example.o',
-                     '-I' + os.path.join(shared.options.binaryen_root, 'src'), '-g', '-L' + libpath, '-pthread']
+                     '-I' + os.path.join(shared.options.binaryen_root, 'src'), '-g', '-pthread']
```
Comment (MaxGraey) [inline check.py:309]: It seems this don't needed due to there are compilation via `-c`, not linking

**Comment by MaxGraey** [comment]
Also after run `./check.py gcc` I got this:
```
[wasm-validator error in function func] local.set's value type must be correct, on 
[none] (local.set $0
 [i64] (i64.const 1234)
)
  (no trace in  c-api-kitchen-sink.txt )
  (no trace in  c-api-kitchen-sink.txt.txt )
```

**Comment by dcodeIO** [comment]
The error seems to be intended as part of `test_nonvalid // create a module that fails to validate`:

https://github.com/WebAssembly/binaryen/blob/daa442b40f92ee5117c9c7c391171c3304abc67e/test/example/c-api-kitchen-sink.c#L1234

**Comment by MaxGraey** [comment]
> The error seems to be intended as part of test_nonvalid // create a module that fails to validate:

Ah, good to know. Thanks!

**Pull request closed without merging**